### PR TITLE
Msi 2030 add ability to mark orders placed with store pickup

### PR DESCRIPTION
**Pull request by swnsma**
### Description (*)
Add possibility to mark orders, created with 'In Store Pickup' Shipping method.
Add extension attribute to order entity with information about related pickup point.

### Fixed Issues

1. magento-engcom/msi#2030: Add an ability to Mark Orders placed with Store Pickup

### Points to discuss
Probably, it will be nice to add this extension attribute to collection join:
```
    <extension_attributes for="Magento\Sales\Api\Data\OrderInterface">
        <attribute code="pickup_point_id" type="string">
            <join reference_table="inventory_pickup_point_order" join_on_field="entity_id" reference_field="order_id">
                <field column="pickup_point_id">pickup_point_id</field>
            </join>
        </attribute>
```
But for this we need to add join processor for Order Repository 'getList' method in Magento 2 Core Repository (https://github.com/magento-engcom/msi/blob/475dc7c32e54621087b9669ca76ab1eacecde12f/app/code/Magento/Sales/Model/OrderRepository.php#L197)
E.g.
```
        /** @var \Magento\Framework\Api\ExtensionAttribute\JoinProcessor $extensionJoinProcessor */
        $extensionJoinProcessor = ObjectManager::getInstance()->get('\Magento\Framework\Api\ExtensionAttribute\JoinProcessor');
        $extensionJoinProcessor->process($searchResult, OrderInterface::class);
```
**UPD**: issue already reported https://github.com/magento/magento2/issues/8035

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento-engcom/msi?pullRequest=2082) <br/>All committers have signed the CLA.

File: app/code/Magento/InventoryInStorePickup/Model/ResourceModel/OrderPickupPoint/GetPickupPointByOrderId.php [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryInStorePickup\Model\ResourceModel\OrderPickupPoint;
+
+use Magento\Framework\App\ResourceConnection;
+
+/**
+ * Class GetPickupPointByOrderId.
```
Comment (ishakhsuvarov) [inline app/code/Magento/InventoryInStorePickup/Model/ResourceModel/OrderPickupPoint/GetPickupPointByOrderId.php]: Repeating class name in the comment is redundant. Clear responsibility description is sufficient.

File: app/code/Magento/InventoryInStorePickup/Model/ResourceModel/OrderPickupPoint/GetPickupPointByOrderId.php [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+ * Get Pickup Point identifier by order identifier.
+ */
+class GetPickupPointByOrderId
+{
+    const ORDER_ID        = 'order_id';
```
Comment (ishakhsuvarov) [inline app/code/Magento/InventoryInStorePickup/Model/ResourceModel/OrderPickupPoint/GetPickupPointByOrderId.php]: Should these constants be `private`?

File: app/code/Magento/InventoryInStorePickup/Model/ResourceModel/OrderPickupPoint/GetPickupPointByOrderId.php [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    /**
+     * Fetch pickup point identifier by order identifier.
+     *
+     * @param int $orderId
+     *
+     * @return string|null
+     */
+    public function execute(int $orderId):?string
```
Comment (ishakhsuvarov) [inline app/code/Magento/InventoryInStorePickup/Model/ResourceModel/OrderPickupPoint/GetPickupPointByOrderId.php]: Why should pickup point id be a `string`? Maybe it's a point code?

File: app/code/Magento/InventoryInStorePickup/etc/db_schema.xml [hunk 4]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0"?>
+<!--
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+-->
+<schema xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:noNamespaceSchemaLocation="urn:magento:framework:Setup/Declaration/Schema/etc/schema.xsd">
+    <table name="inventory_pickup_point_order" resource="default" engine="innodb">
```
Comment (ishakhsuvarov) [inline app/code/Magento/InventoryInStorePickup/etc/db_schema.xml]: We have to discuss `Pickup Point` naming. It's best to align it with business domain

File: app/code/Magento/InventoryInStorePickup/etc/module.xml [hunk 5]
```diff
@@ -6,5 +6,10 @@
  */
 -->
 <config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:Module/etc/module.xsd">
-    <module name="Magento_InventoryInStorePickup" setup_version="1.0.0" />
+    <module name="Magento_InventoryInStorePickup" setup_version="1.0.0">
+        <sequence>
+            <module name="Magento_Sales" />
+            <module name="Magento_Inventory" />
```
Comment (ishakhsuvarov) [inline app/code/Magento/InventoryInStorePickup/etc/module.xml]: Should it be `Magento_Inventory` or `InventoryApi`?

Also `sales` should be added to composer dependencies

**Review by ishakhsuvarov — CHANGES_REQUESTED** [review]
* `Pickup Point` term has to be discussed and aligned with business domain
* Please add integration tests as well
* Please update with the latest changes from `store-pickup` branch

File: app/code/Magento/InventoryInStorePickup/Model/ResourceModel/OrderPickupPoint/GetPickupPointByOrderId.php (hunk 1 above)
Comment (swnsma) [reply app/code/Magento/InventoryInStorePickup/Model/ResourceModel/OrderPickupPoint/GetPickupPointByOrderId.php]: Done.

File: app/code/Magento/InventoryInStorePickup/Model/ResourceModel/OrderPickupPoint/GetPickupPointByOrderId.php (hunk 2 above)
Comment (swnsma) [reply app/code/Magento/InventoryInStorePickup/Model/ResourceModel/OrderPickupPoint/GetPickupPointByOrderId.php]: Done.

File: app/code/Magento/InventoryInStorePickup/etc/module.xml (hunk 5 above)
Comment (swnsma) [reply app/code/Magento/InventoryInStorePickup/etc/module.xml]: For now, dependency on Magento_Sales only.
Added to composer.json.

File: app/code/Magento/InventoryInStorePickup/Model/ResourceModel/OrderPickupPoint/GetPickupPointByOrderId.php (hunk 3 above)
Comment (swnsma) [reply app/code/Magento/InventoryInStorePickup/Model/ResourceModel/OrderPickupPoint/GetPickupPointByOrderId.php]: Fixed.

File: app/code/Magento/InventoryInStorePickup/etc/db_schema.xml (hunk 4 above)
Comment (swnsma) [reply app/code/Magento/InventoryInStorePickup/etc/db_schema.xml]: Changed to 'Pickup Location'.

File: app/code/Magento/InventoryInStorePickup/Test/_files/create_in_store_pickup_quote_on_eu_website.php [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+$cart->setCustomerEmail('admin@example.com');
+$cart->setCustomerIsGuest(true);
+$store = $storeRepository->get('store_for_eu_website');
+$cart->setStoreId($store->getId());
+$storeManager->setCurrentStore($store->getCode());
+
+/** @var AddressInterface $address */
+$address = $addressFactory->create(
+    [
+        'data' => [
```
Comment (ishakhsuvarov) [inline app/code/Magento/InventoryInStorePickup/Test/_files/create_in_store_pickup_quote_on_eu_website.php:38]: I love how you save a US address to the Quote in fixture designed for EU Website :)

**Review by ishakhsuvarov — APPROVED** [review]

**Pull request merged**